src/functional/ffront/type_info.py

```diff
@@ -385,6 +385,37 @@ def promote_dims(
     return topologically_sorted_list
 
 
+def promote_to_mask_type(
```

Comment:
Since the semantic meaning of this function and the terms used in it's code and docstring are quite limited to the only place it is being used, it should probably be moved to `type_deduction.py` until it turns out to be more widely useful.